crates/oxc_linter/src/rules/promise/no_callback_in_promise.rs

```diff
@@ -0,0 +1,193 @@
+use oxc_ast::{
+    ast::{Expression, MemberExpression},
+    AstKind,
+};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{GetSpan, Span};
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+static CALLBACKS: [&str; 4] = ["done", "cb", "callback", "next"];
+
+fn no_callback_in_promise_diagnostic(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("Avoid calling back inside of a promise").with_label(span)
```

Comment:
Could we add some help text here to tell users how to fix this problem? Like using `setTimeout` to wrap the callback

Reply:
I found that they haven't achieved this since 2021. Recently, there is [a new PR](https://github.com/eslint-community/eslint-plugin-promise/pull/514/files) working on this, and I will try to fix it.
```
  // TODO: support safe callbacks (see #220)
  // 'whatever.then((err) => { process.nextTick(() => cb()) })',
  // 'whatever.then((err) => { setImmediate(() => cb())) })',
  // 'whatever.then((err) => setImmediate(() => cb()))',
  // 'whatever.then((err) => process.nextTick(() => cb()))',
  // 'whatever.then((err) => process.nextTick(cb))',
  // 'whatever.then((err) => setImmediate(cb))',
```

Reply:
There are also some issues with this PR, let's first implement the current feature.